crates/core/tedge_mapper/src/collectd_mapper/collectd.rs

```diff
@@ -193,15 +208,61 @@ mod tests {
             metric_key,
             timestamp,
             metric_value,
-        } = collectd_message;
-
+        } = collectd_message.index(0);
         assert_eq!(metric_group_key, "temperature");
+
         assert_eq!(metric_key, "value");
         assert_eq!(
+            *timestamp,
+            Utc.ymd(1973, 11, 29).and_hms_milli(21, 33, 09, 0)
+        );
+        assert_eq!(*metric_value, 32.5);
+    }
+
+    #[test]
+    fn collectd_message_parsing_multi_valued_measurement() {
+        let topic = Topic::new("collectd/localhost/temperature/value").unwrap();
+        let mqtt_message = Message::new(&topic, "123456789:32.5:45.2");
+
+        let collectd_message = CollectdMessage::parse_from(&mqtt_message).unwrap();
+
+        let CollectdMessage {
+            metric_group_key,
+            metric_key,
+            timestamp,
+            metric_value,
+        } = collectd_message.index(0);
+        assert_eq!(metric_group_key, "temperature");
+
+        assert_eq!(metric_key, "value_val1");
+        assert_eq!(
+            *timestamp,
+            Utc.ymd(1973, 11, 29).and_hms_milli(21, 33, 09, 0)
+        );
+
+        assert_eq!(*metric_value, 32.5);
+        assert_eq!(collectd_message.index(0).metric_group_key, "temperature");
+        assert_eq!(collectd_message.index(0).metric_key, "value_val1");
+        assert_eq!(
+            collectd_message.index(0).timestamp,
+            Utc.ymd(1973, 11, 29).and_hms_milli(21, 33, 09, 0)
+        );
+        assert_eq!(collectd_message.index(0).metric_value, 32.5);
```

Comment:
These assertions duplicate what is already checked lines 229 .. 243

Reply:
good catch removed redundant check.